File: jdroid-android-about/src/main/java/com/jdroid/android/about/AboutContext.java
package com.jdroid.android.about;

import com.jdroid.android.context.AbstractAppContext;
import com.jdroid.android.fragment.AbstractFragment;
import com.jdroid.android.utils.AppUtils;
import com.jdroid.java.utils.ReflectionUtils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class AboutContext extends AbstractAppContext {

	@NonNull
	@SuppressWarnings("unchecked")
	public Class<? extends AbstractFragment> getAboutFragmentClass() {
		return (Class<? extends AbstractFragment>)ReflectionUtils.getClass("com.jdroid.android.about.AboutFragment");
	}

	@NonNull
	@SuppressWarnings("unchecked")
	public Class<? extends AbstractFragment> getLibrariesFragmentClass() {
		return (Class<? extends AbstractFragment>)ReflectionUtils.getClass("com.jdroid.android.about.LibrariesFragment");
	}

	@Nullable
	public Class<? extends AbstractFragment> getSpreadTheLoveFragmentClass() {
		return null;
	}

	public boolean isBetaTestingEnabled() {
		return false;
	}

	public String getBetaTestingUrl() {
		return "https://play.google.com/apps/testing/" + AppUtils.INSTANCE.getReleaseApplicationId();
	}
}

File: jdroid-android-about/src/main/java/com/jdroid/android/about/AboutContext.kt
package com.jdroid.android.about

import com.jdroid.android.context.AbstractAppContext
import com.jdroid.android.fragment.AbstractFragment
import com.jdroid.android.utils.AppUtils
import com.jdroid.java.utils.ReflectionUtils

open class AboutContext : AbstractAppContext() {

    @Suppress("UNCHECKED_CAST")
    fun getAboutFragmentClass(): Class<out AbstractFragment> {
        return ReflectionUtils.getClass("com.jdroid.android.about.AboutFragment") as Class<out AbstractFragment>
    }

    @Suppress("UNCHECKED_CAST")
    fun getLibrariesFragmentClass(): Class<out AbstractFragment> {
        return ReflectionUtils.getClass("com.jdroid.android.about.LibrariesFragment") as Class<out AbstractFragment>
    }

    open fun getSpreadTheLoveFragmentClass(): Class<out AbstractFragment>? {
        return null
    }

    open fun isBetaTestingEnabled(): Boolean {
        return false
    }

    open fun getBetaTestingUrl(): String {
        return "https://play.google.com/apps/testing/" + AppUtils.getReleaseApplicationId()
    }
}
